sw/device/silicon_creator/manuf/base/ft_personalize.c

```diff
@@ -308,12 +308,22 @@ static status_t hash_certificate(const flash_ctrl_info_page_t *page,
                                  size_t offset, size_t *size) {
   uint8_t buffer[1024];  // 1K should be enough for the largest certificate.
   uint32_t cert_size;
-  uint32_t bytes_read;
+  uint16_t wrapped_cert_size;
+  uint16_t name_len;
+  uint16_t hdr_and_name_len;
 
   // Read the first word of the certificate which contains it's size.
   TRY(flash_ctrl_info_read(page, offset, 1, buffer));
-  bytes_read = sizeof(uint32_t);
-  cert_size = cert_x509_asn1_decode_size_header(buffer);
+  perso_tlv_cert_header_t crth;
+  memcpy(&crth, &buffer[0], sizeof(crth));
```

Comment:
```suggestion
  memcpy(&crth, buffer, sizeof(crth));
```

Maybe we should also add a `static_assert()` some where that checks that `sizeof(crth) <= 4`, since we only read 1 word from the flash page on line 316 above (which is most efficient)?